source/plugins/gloperate-glraw-loaders/FileNameSuffix.cpp

```diff
@@ -0,0 +1,159 @@
+#include <FileNameSuffix.h>
+
+#include <cassert>
+#include <regex>
+#include <string>
+
+#include <glbinding/gl/gl.h>
+
+
+namespace
+{
+    static const std::map<std::string, gl::GLenum> formatsBySuffix =
+	{
+        { "rh",     gl::GL_RED      },
+        { "g",      gl::GL_GREEN    },
+        { "b",      gl::GL_BLUE     },
+        { "rg",     gl::GL_RG       },
+        { "rgb",    gl::GL_RGB      },
+        { "bgr",    gl::GL_BGR      },
+        { "rgba",   gl::GL_RGBA     },
+        { "bgra",   gl::GL_BGRA     }
+	};
+
+    static const std::map<std::string, gl::GLenum> typesBySuffix =
+	{
+        { "ub", gl::GL_UNSIGNED_BYTE    },
+        { "b",  gl::GL_BYTE             },
+        { "us", gl::GL_UNSIGNED_SHORT   },
+        { "s",  gl::GL_SHORT            },
+        { "ui", gl::GL_UNSIGNED_INT     },
+        { "i",  gl::GL_INT              },
+        { "f",  gl::GL_FLOAT            }
+
+    #ifdef gl::GL_ARB_texture_compression_rgtc
```

Comment:
This doesn't work as expected. Remove the ```#idef``` conditional code and add the formats in each case.